library/alloc/src/collections/binary_heap/mod.rs

```diff
@@ -1003,6 +1003,39 @@ impl<T, A: Allocator> BinaryHeap<T, A> {
         self.data.get(0)
     }
 
+    /// Returns the smallest item in the binary heap, or `None` if it is empty.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use std::collections::BinaryHeap;
+    /// let mut heap = BinaryHeap::new();
+    /// assert_eq!(heap.peek_end(), None);
+    ///
+    /// heap.push(1);
+    /// heap.push(5);
+    /// heap.push(2);
+    /// assert_eq!(heap.peek_end(), Some(&1));
+    ///
+    /// ```
+    ///
+    /// # Time complexity
+    ///
+    /// Cost is *O*(1) in the worst case.
+    #[must_use]
+    #[stable(feature = "rust1", since = "1.76.0")]
+    pub fn peek_end(&self) -> Option<&T> {
+        let len = self.data.len();
+        if len > 0 {
+            // SAFETY: Just checked that the index must exist.
+            unsafe { self.data.get_unchecked(len - 1) }
```

Comment:
Would `data.last()` also work here but be safe?

Reply:
Yep, fixed